gtfs-validator/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/RouteHasNoDuplicateStopsValidatorTest.java

```diff
@@ -0,0 +1,191 @@
+                                             .setStopSequence(1)
+                                             .build();
+  private final GtfsStopTime stopTimeTrip3_Stop2 = new GtfsStopTime.Builder()
+                                             .setCsvRowNumber(16)
+                                             .setTripId("trip3")
+                                             .setStopId("1002")
+                                             .setStopSequence(2)
+                                             .build();                                                                                     
+  private final GtfsStop stopId1001 =
+      new GtfsStop.Builder().setCsvRowNumber(3).setStopId("1001").setStopName("stop1").build();
+  private final GtfsStop stopId1002 =
+      new GtfsStop.Builder().setCsvRowNumber(4).setStopId("1002").setStopName("stop1").build();
+  private final GtfsStop stopId1003 =
+      new GtfsStop.Builder().setCsvRowNumber(18).setStopId("1003").setStopName("stop3").build();    
+  
+  @Test
+  public void routeWithDuplicateShouldGenerateNotice() {
+    final NoticeContainer noticeContainer = new NoticeContainer();
+    RouteHasNoDuplicateStopsValidator validator = new RouteHasNoDuplicateStopsValidator();
+
+    // Create tripTable:
+    validator.tripTable =
+        GtfsTripTableContainer.forEntities(Arrays.asList(trip1_route1), noticeContainer);
+
+    // Create stopTimeTable:
+    validator.stopTimeTable = GtfsStopTimeTableContainer.forEntities(
+        Arrays.asList(stopTimeTrip1_Stop1, stopTimeTrip1_Stop2), noticeContainer);
+
+    // Create stopTable:
+    validator.stopTable =
+        GtfsStopTableContainer.forEntities(Arrays.asList(stopId1001, stopId1002), noticeContainer);
+    
+    validator.validate(noticeContainer);
+    assertThat(noticeContainer.getNotices())
+        .contains(new RouteWithDuplicateStopNotice(/* stopName = */ "stop1",
+            /* prevStopId  = */ "1001", /* prevCsvRowNumber = */ 3, /* stopId  = */ "1002",
+            /* csvRowNumber = */ 4, /* routeId = */ "route1"));
+  }
+  
+  @Test
+  public void routeWithRepeatedDuplicateShouldGenerateOneNotice() {
+    final NoticeContainer noticeContainer = new NoticeContainer();
+    RouteHasNoDuplicateStopsValidator validator = new RouteHasNoDuplicateStopsValidator();
+
+    // Create tripTable:
+    validator.tripTable =
+        GtfsTripTableContainer.forEntities(Arrays.asList(trip1_route1, trip2_route1), noticeContainer);
+
+    // Create stopTimeTable:
+    validator.stopTimeTable = GtfsStopTimeTableContainer.forEntities(
+        Arrays.asList(stopTimeTrip1_Stop1, stopTimeTrip1_Stop2, stopTimeTrip2_Stop1, stopTimeTrip2_Stop2), noticeContainer);
+
+    // Create stopTable:
+    validator.stopTable =
+        GtfsStopTableContainer.forEntities(Arrays.asList(stopId1001, stopId1002), noticeContainer);
+    
+    validator.validate(noticeContainer);
+    System.out.println(noticeContainer.getNotices());
```

Comment:
Do you need to print out the notice container? Or was this line just left over from debugging?

Reply:
Sorry yes, should have cleaned this up before. Gone now.